mlaut/benchmarking/base.py

```diff
@@ -0,0 +1,179 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+__author__ = ["Markus Löning", "Viktor Kazakov"]
+__all__ = ["BaseDataset", "HDDBaseDataset", "BaseResults", "HDDBaseResults"]
+
+import os
+from joblib import dump
+from joblib import load
+from warnings import warn
+
+
+class BaseDataset:
+
+    def __init__(self, name):
+        self._name = name
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
+        return f"{class_name}(name={self.name})"
+
+    def load(self):
+        raise NotImplementedError()
+
+    @property
+    def name(self):
+        return self._name
+
+
+class HDDBaseDataset(BaseDataset):
+
+    def __init__(self, path, name):
+        self._path = path
+        super(HDDBaseDataset, self).__init__(name=name)
+
+    @property
+    def path(self):
+        return self._path
+
+    @staticmethod
+    def _validate_path(path):
+        """Helper function to validate paths"""
+        # check if path already exists
+        if not os.path.exists(path):
+            raise ValueError(f"No dataset found at path: {path}")
+
+
+class BaseResults:
+
+    def __init__(self):
+        # assigned during fitting of orchestration
+        self.strategy_names = []
+        self.dataset_names = []
+        self.cv = None
+
+    def save_predictions(self, strategy_name, dataset_name, y_true, y_pred, y_proba, index, cv_fold,
+                         train_or_test):
+        raise NotImplementedError()
+
+    def load_predictions(self, cv_fold, train_or_test):
+        """Loads predictions for all datasets and strategies iteratively"""
+        raise NotImplementedError()
```

Comment:
```suggestion
        raise NotImplementedError("abstract method")
```

for the other methods below as well, just to make clear what it is